cmd/bucket-listobjects-handlers.go

```diff
@@ -193,6 +196,91 @@ func (api objectAPIHandlers) ListObjectsV2MHandler(w http.ResponseWriter, r *htt
 	writeSuccessResponseXML(w, encodeResponse(response))
 }
 
+func listObjectsV2InZip(ctx context.Context, objectAPI ObjectLayer, bucket, prefix, token, delimiter string, maxKeys int, fetchOwner bool, startAfter string) (ListObjectsV2Info, error) {
+
+	idx := strings.Index(prefix, archivePattern)
+	idx += len(archivePattern)
+	zipPath := prefix[:idx-len(slashSeparator)]
+
+	zipObjInfo, err := objectAPI.GetObjectInfo(ctx, bucket, zipPath, ObjectOptions{})
+	if err != nil {
+		// Return empty listing
+		return ListObjectsV2Info{}, nil
+	}
+
+	var zipInfo []byte
+
+	if z, ok := zipObjInfo.UserDefined[zipInfoMetadataName]; ok {
```

Comment:
Move this to its own file so that its easy to add tests - such as `bucket-listobjects-zip.go` ?